sceptre/diffing/stack_differ.py

```diff
@@ -167,12 +167,15 @@ def _extract_parameters_from_generated_stack(self, stack: Stack) -> dict:
         """
         formatted_parameters = {}
         for key, value in stack.parameters.items():
+            # When boto3 receives "None" for a cloudformation parameter, it treats it as if the
+            # value is not passed at all. To be consistent in our diffing, we need to skip Nones
+            # altogether.
+            if value is None:
+                continue
+
             if isinstance(value, list):
                 value = ",".join(item.rstrip("\n") for item in value)
-            if isinstance(value, str):
-                formatted_parameters[key] = value.rstrip("\n")
-            elif value is None:
-                formatted_parameters[key] = ""
+            formatted_parameters[key] = value.rstrip("\n")
```

Comment:
You might want to consider maybe raising some sort of value error that is readable if somebody does happen to specify a parameter value that isn't a str, list, or None. Something like "Parameter xyz is not a string; This is not a valid parameter value". 

I get the sense that the bad error messaging is 50% of the problem this issue raises.